### PR TITLE
feat(kprompt): add kprompt kongponent

### PR DESCRIPTION
**Pull request by kaiarrowood**
### Summary

Add new `KPrompt` dialog for modals that prompt a user to take an action.

Warning:

image

Danger:

image

#### Changes made:
* Add new kongponent
* Add tests
* Add docs
* Manually set version of `KTableLegacy` to match other kongponents versions

### Vue 3 Upgrade

**If your PR contains code that needs to be ported to the `next` branch, please add the [`port to next branch`](https://github.com/Kong/kongponents/labels/port%20to%20next%20branch) label to your PR.**

We are currently in the process of upgrading Kongponents to Vue 3. If changes are made to a component or doc file on the `main` branch, a corresponding PR needs to be made into the `next` branch that includes:

- The component feature/fix, updated for Vue 3 and the Composition API.
- Documentation updates for the component changes, as well as updating examples and usage to Vue 3 and the Composition API.
- Updates to the corresponding `.spec.ts` test file(s) to utilize [Cypress Component Testing](https://docs.cypress.io/guides/component-testing/introduction).

If you have questions, tag `@adamdehaven` or `@kaiarrowood`.

**Does your PR modify a component [that already exists on the `next` branch](https://github.com/Kong/kongponents/tree/next/src/components)?**

  - [ ] **Yes**, and there is a corresponding PR to update the component on the `next` branch
    - `LINK_TO_PR_ON_NEXT_BRANCH` (**required**)
  - [x] **No**, the component does not yet exist on `next` branch.

### PR Checklist
- [x] Does not introduce dependencies
- [x] **Functional:** all changes do not break existing APIs and if so, bump major version.
- [x] **Tests pass:** check the output of yarn test packages/<Kongponent>
- [x] **Naming:** the files and the method and prop variables use the same naming conventions as other Kongponents
- [x] **Framework style:** abides by the essential rules in Vue's style guide
- [x] **Cleanliness:** does not have formatting issues, unused code (e.g., console.logs), or leftover comments
- [x] **Docs:** includes a technically accurate README, uses JSDOC where appropriate

### CODE REVIEW

File: packages/KPrompt/KPrompt.vue [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+          </slot>
+
+          <div
+            v-if="confirmationText"
+            class="k-prompt-confirm-text">
+            Type "<span class="bold-600">{{ confirmationText }}</span>" to confirm your action.
+
+            <KInput
+              v-model="confirmationInput"
+              class="pt-2" />
```
Comment (adamdehaven) [inline packages/KPrompt/KPrompt.vue:48]: Add `w-100` class so the input is full width

Comment (kaiarrowood) [reply packages/KPrompt/KPrompt.vue:48]: Good catch, fixed!

File: packages/KPrompt/KPrompt.vue [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+            {{ message }}
+          </slot>
+
+          <div
+            v-if="confirmationText"
+            class="k-prompt-confirm-text w-100">
+            Type "<span class="bold-600">{{ confirmationText }}</span>" to confirm your action.
+
+            <KInput
+              v-model.trim="confirmationInput"
```
Comment (adamdehaven) [inline packages/KPrompt/KPrompt.vue:47]: TL;DR change back to `v-model="confirmationInput"`

I thought about this and actually, I think we should **REMOVE** the `.trim` -- we want them to type the exact text, so extra spaces _should_ prevent submission.

File: packages/KPrompt/KPrompt.vue [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+          <KButton
+            appearance="outline"
+            class="k-prompt-cancel mr-2"
+            @click="close"
+            @keyup.esc="close">{{ cancelButtonText }}</KButton>
+          <KButton
+            :appearance="type === 'danger' ? 'danger' : 'primary'"
+            :disabled="disableProceedButton"
+            class="k-prompt-proceed"
+            @keyup.enter="proceed"
```
Comment (adamdehaven) [inline packages/KPrompt/KPrompt.vue:66]: Pressing `Enter` on the docs page does not trigger the `proceed` action or close the prompt for me.

File: packages/KPrompt/KPrompt.vue (hunk 2 above)
Comment (adamdehaven) [inline packages/KPrompt/KPrompt.vue:47]: When on the docs page, if I type in the `I Agree` text into the Prompt, then click `OK` to proceed, the modal closes correctly; however, if I open the same prompt again, the `I Agree` text is already inside of the `KInput`

image

We should clear the value of `confirmationInput` before the open action actually shows the Prompt.

**Review by adamdehaven — CHANGES_REQUESTED** [review]

File: packages/KPrompt/KPrompt.vue [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+    title: {
+      type: String,
+      default: ''
+    },
+    type: {
+      type: String,
+      default: 'information',
+      validator: (value) => {
+        return [
+          'information',
```
[the remark this answers is not in the document]
Comment (kaiarrowood) [reply packages/KPrompt/KPrompt.vue]: Done

File: packages/KPrompt/KPrompt.vue [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+          'warning',
+          'danger'
+        ].indexOf(value) !== -1
```
[the remark this answers is not in the document]
Comment (kaiarrowood) [reply packages/KPrompt/KPrompt.vue]: Done

File: docs/components/prompt.md [hunk 6]
```diff
@@ -0,0 +1,247 @@
...
+  @proceed="buttonsIsOpen = false"
+/>
+```
+
+### Variations
+
+- `type` - Can be `danger`, `warning`, or `information`. Defaults to `information`
+- `confirmationText` - Provide a string the user must type before the action button becomes enabled
+
+Use the `information` prompt type to notify the user abbout general information associated with the action about
```
[the remark this answers is not in the document]
Comment (kaiarrowood) [reply docs/components/prompt.md]: Fixed

File: packages/KPrompt/KPrompt.vue [hunk 7]
```diff
@@ -0,0 +1,224 @@
...
+      }
+    },
+    message: {
+      type: String,
+      default: 'Body content'
```
[the remark this answers is not in the document]
Comment (kaiarrowood) [reply packages/KPrompt/KPrompt.vue]: Fixed.

File: packages/KPrompt/KPrompt.vue [hunk 8]
```diff
@@ -0,0 +1,224 @@
+<template>
+  <KModal
+    :is-visible="isVisible"
+    :title="displayTitle"
+    class="k-prompt"
+    @keyup.esc="close">
```
[the remark this answers is not in the document]
Comment (kaiarrowood) [reply packages/KPrompt/KPrompt.vue]: Fixed

File: packages/KPrompt/KPrompt.vue [hunk 9]
```diff
@@ -0,0 +1,224 @@
...
+    displayTitle () {
+      if (this.title) {
+        if (this.type === 'warning') {
+          return 'Warning: ' + this.title
+        }
+
+        return this.title
+      }
+
+      return this.capitalize(this.type)
```
[the remark this answers is not in the document]
Comment (kaiarrowood) [reply packages/KPrompt/KPrompt.vue:144]: Done

File: packages/KPrompt/KPrompt.vue (hunk 2 above)
Comment (kaiarrowood) [reply packages/KPrompt/KPrompt.vue]: Done

File: packages/KPrompt/KPrompt.vue (hunk 3 above)
Comment (kaiarrowood) [reply packages/KPrompt/KPrompt.vue]: Fixed, but requires you to click on the dialog first.

File: packages/KPrompt/KPrompt.vue (hunk 2 above)
Comment (kaiarrowood) [reply packages/KPrompt/KPrompt.vue]: Fixed.

File: docs/components/prompt.md [hunk 10]
```diff
@@ -0,0 +1,247 @@
...
+    return {
+      defaultIsOpen: false
+    }
+  }
+}
+```
+
+## Props
+
+### Functional
```
[the remark this answers is not in the document]
Comment (kaiarrowood) [reply docs/components/prompt.md]: Done

**Review by adamdehaven — APPROVED** [review]
LGTM 🎉

**Pull request merged**